samples/bluetooth/alexa_gadget/CMakeLists.txt

```diff
@@ -0,0 +1,25 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+cmake_minimum_required(VERSION 3.13.1)
+
+# Set ZDebug as the default build
+if (NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE ZDebug)
+endif()
```

Comment:
I this a sample or should it actually be location under `applications` like nRF Desktop and asset tracker ?

We don't use the ZDebug and ZRelease in the samples. (This PR would be the first)

Reply:
I did discuss the sample vs application aspect with Joakim, and the preliminary conclusion was that it is closer to a sample than an application. Although it is a fairly elaborate sample, mainly due to the complexity of this particular Service and Profile.
The reason for using ZDebug and ZRelease is due to a desire to show compatibility with nRF52810, which severely limits the logging and debugging features that can be used. Perhaps it would make sense to rename these to something along the lines of "ZWithLogging" and "ZWithoutLogging"

Reply:
> The reason for using ZDebug and ZRelease is due to a desire to show compatibility with nRF52810, which severely limits the logging and debugging features that can be used. 

In that case you should provide a nRF52810 overlay config that can be automatically used for this board.
Place this file `boards/nrf52dk_nrf52810.conf` in `samples/bluetooth/alexa_gadget` and it will be automatically loaded together with `prj.conf` when building for that board, and then you can adjust log settings.

Let's use what Zephyr already provides in our samples.

Reply:
Yes, good idea regarding the overlay. Will do that instead

Reply:
it seems to me that all the files discovered by the globbing here are the same file names found here, just with another file extension:
https://github.com/nrfconnect/sdk-nrf/blob/7f57514153468ee33a48495e465ef0fb1ba13e5e/samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt#L155-L197

So there is no reason for the globbing.
You can make a list of files, and then for example use a `list(TRANSFORM ...)` to change the file extension:
https://cmake.org/cmake/help/latest/command/list.html#transform

Other options are possible as well, but no reason for globbing.